### PR TITLE
Enable Javadoc-related Checkstyle checks

### PR DESCRIPTION
**Pull request by basil**
Enables various Javadoc-related Checkstyle checks and fixes all existing violations.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by daniel-beck** [comment]
Is it weird that GitHub only finds 11 Maven POMs with `AtclauseOrder`?

**Comment by timja** [comment]
> AtclauseOrder

? https://github.com/search?l=XML&q=AtclauseOrder&type=Code

> 19,326 code results

**Comment by daniel-beck** [comment]
@timja Thanks, I guess that's a "no" then. I checked https://github.com/search?l=Maven+POM&q=AtclauseOrder&type=Code

**Comment by timja** [comment]
> @timja Thanks, I guess that's a "no" then. I checked [github.com/search?l=Maven+POM&q=AtclauseOrder&type=Code](https://github.com/search?l=Maven+POM&q=AtclauseOrder&type=Code)

see my update to explain more :)

**Comment by basil** [comment]
> FWIW I wonder whether putting `@deprecated` last (rather than just going alphabetical) would match our existing practices better. And before that, `@since`. Semantically they are different from the rest IMO, so that would make sense.

That sounds fine to me; I have no strong preference. I wanted to keep this change minimal so I intentionally left out `@since` so that I didn't have to fix dozens of violations. Perhaps we can improve this in subsequent PRs.

File: pom.xml
Comment (daniel-beck) [inline pom.xml:485]: As general order, this looks good:
1. `param`
2. `return`
3. `throws`
4. `deprecated`
👍 

A bit surprised to see `@exception` at the end rather than next to `@throws`, perhaps we could just prohibit that one somehow? And `@version` doesn't look particularly useful either?

**Review by daniel-beck — APPROVED** [review]

File: pom.xml
Comment (basil) [reply pom.xml]: Sure, I changed this to `@param, @return, @throws, @deprecated`.

Comment (daniel-beck) [reply pom.xml]: To clarify, I liked the order as written :) Even without any `@serial*` in the current code, that was fine.

**Pull request merged**
